ingraph-engine/ingraph-engine-ingraph-ire/src/main/scala/ingraph/ire/Indexer.scala

```diff
@@ -16,6 +18,11 @@ case class IngraphVertex(id: Long,
   val reverseEdges: mutable.ListMap[String, IngraphEdge] = mutable.ListMap[String, IngraphEdge]()
   override def toString: String = s"Vertex($id, $properties)"
 }
+object IngraphVertex {
+  implicit val ingraphObject = new IngraphObject[IngraphVertex] {
```

Comment:
The benefit is that it is extendible (no need to modify a class to add new traits implementations).
The tradeoff is sadly the syntax :(